front/src/components/TodoDialog.js

```diff
@@ -0,0 +1,174 @@
+import React, { useState } from 'react';
+import { 
+    Button,
+    Dialog, 
+    DialogActions,
+    DialogContent, 
+    DialogTitle, 
+    TextField, 
+    Box,
+    Snackbar, 
+    Alert } from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import axios from "axios"; 
+
+export default function BookmarkDialog({getData}){
+    const [open, setOpen] = useState(false)
+    const [snackOpen, setSnackOpen] = useState(false)
+
+    const form = {
+        title: "",
+        description: "",
+        date: "",
+        assigned: ""
+    }
+
+    const snackbar = {
+        text: "",
+        severity: ""
+    }
+
+    const [formValue, setFormValue] = useState(form)
+    const [snackbarInfo, setSnackbarInfo] = useState(snackbar)
+
+    const handleFormChange = e => {
+        const { value, name } = e.target
+        const newValue = {
+            ...formValue,
+            [name]: value
+        }
+        setFormValue(newValue)
+    }
+
+    const handleClickOpen = () => {
+        setOpen(true)
+    }
+
+    const handleClose = () => {
+        setOpen(false)
+    }
+
+    const handleSnackClose = () => {
+        setSnackOpen(false)
+    }
+
+    const handleSubmit = e => {
+        e.preventDefault()
+        if(formValue.title && formValue.description && formValue.date && formValue.assigned){
```

Comment:
Add else statement with helper text to inform user of missing fields

Reply:
done